pyradiosky/skymodel.py

```diff
@@ -4716,6 +4714,283 @@ def read_idl_catalog(
             run_check_acceptability=run_check_acceptability,
         )
 
+    def read(
+        self,
+        filename,
+        filetype=None,
+        run_check=True,
+        check_extra=True,
+        run_check_acceptability=True,
+        # Gleam vot
+        spectral_type=None,
+        with_error=False,
+        # fhd
+        expand_extended=True,
+        # vot
+        table_name=None,
+        id_column=None,
+        ra_column=None,
+        dec_column=None,
+        flux_columns=None,
+        reference_frequency=None,
+        freq_array=None,
+        spectral_index_column=None,
+        flux_error_columns=None,
+        history="",
```

Comment:
I'm not sure it really makes sense to specify these here. Might as well make use of `**kwargs`, and point people to the docs for each of the sub-methods for specific inputs.

Reply:
Hmm. I think it's much easier for the users to have all the parameters defined for the function they're trying to call, rather than having to iterate through the docs to find all the options. Also, I think that would tend to push users away from defaulting to using this generic method, which is the direction I want to go. And I'd still have to have the code to parse out which things to pass to each specific read because most of these are only accepted by one of the specific method. I did recently figure out how to make other sub headings for the docs, so I could break out the file-type specific keywords that way if you think that'd be easier to understand for users (I'm working on a PR for that in pyuvdata now).

Reply:
I get what you mean, and I am generally sympathetic. However, to me, this seems like a lot of parameters to reproduce in *three* places, where in two of them you could just have `**kwargs` and reference the sub-functions in the doc string. You wouldn't have to parse, because kwargs should only have the correct arguments in it for any specific sub-function anyway (i.e. whatever was actually passed by the user). 

It also means that if any parameter is added to eg. `read_vot` then that parameter has to be added to two other places, which increases risk of mistakes in the future. In the end, I'll respect your opinion here, but I would still lean towards using kwargs in this instance.

Reply:
We discussed this on the telecon and the consensus was that we wanted to keep this explicit for users at the expense of a higher maintenance burden. However, to make it more readable, we will add more section headings for the different types of parameters, similar to what we are doing on pyuvdata: https://github.com/RadioAstronomySoftwareGroup/pyuvdata/pull/1158

Reply:
OK :-)